src/legacy/integrations/contact-input-integration-wrapper.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React, { useMemo } from 'react';
+
+import {
+	type ContactChipAction,
+	ContactInput,
+	ContactInputItem,
+	type ContactInputProps
+} from './contact-input';
+import { useActionEditDL } from '../../actions/edit-dl';
+import { DistributionList } from '../../model/distribution-list';
+
+export type ContactInputIntegrationWrapperProps = Omit<ContactInputProps, 'contactActions'> & {
+	actions?: ContactInputProps['contactActions'];
+};
+
+const isDistributionList = (
+	contact: DistributionList | ContactInputItem
+): contact is DistributionList => 'isOwner' in contact;
+
+export const ContactInputIntegrationWrapper = ({
+	actions,
+	...rest
+}: ContactInputIntegrationWrapperProps): React.JSX.Element => {
+	const actionEditDL = useActionEditDL();
+
+	const defaultActions = useMemo(
+		(): Array<ContactChipAction> => [
+			{
+				id: `chip-action-${actionEditDL.id}`,
+				label: actionEditDL.label,
+				type: 'button',
+				icon: actionEditDL.icon,
+				isVisible: (contact: DistributionList | ContactInputItem) =>
+					isDistributionList(contact) && contact.isOwner,
+				onClick: (contact: DistributionList | ContactInputItem): void => {
+					isDistributionList(contact) &&
```

Comment:
is this check needed? if yes, isOwner check is required too?

Reply:
I think we missed using the canExecute of the action in the isVisible and we used the isOwner directly. This is also why you found that the canExecute is not used.

Reply:
The check has been updated by calling the `canExecute` function